core/src/client/async_client/mod.rs

```diff
@@ -186,16 +208,10 @@ impl ClientT for Client {
 		let mut sender = self.to_back.clone();
 		let fut = sender.send(FrontToBack::Notification(raw));
 
-		let timeout = tokio::time::sleep(self.request_timeout);
-
-		let res = tokio::select! {
-			x = fut => x,
-			_ = timeout => return Err(Error::RequestTimeout)
-		};
-
-		match res {
-			Ok(()) => Ok(()),
-			Err(_) => Err(self.read_error_from_backend().await),
+		match future::select(fut, Delay::new(self.request_timeout)).await {
+			Either::Left((Ok(()), _)) => Ok(()),
+			Either::Left((Err(_), _)) => Err(self.read_error_from_backend().await),
+			Either::Right((_, _)) => Err(Error::RequestTimeout),
```

Comment:
To me this looks like it does exactly the same thing as before but this looks more readable. Is that the only reason you switched away from `select!`?

Reply:
yes because `tokio` doesn't compile for wasm AFAIU but maybe the macros does?!